### ISSUE TITLE
cmd/link: support internal linking for Android and iOS

### ISSUE THREAD

**Issue by eliasnaur** [golang/go#31819]
We're about to run self-hosted Android and iOS builders (#31722). It would be nice if Go could produce internally linked binaries for those platforms, for a less hacky bootstrapping and for being able to build x/build/cmd/buildlet and stage0 everywhere. Depends on #31343 Android.

**Comment by bcmills** [comment]
> Depends on https://github.com/golang/go/issues/31343 Android.

#31343 is marked as fixed. Is there anything else preventing the use of internal linking on Android?
(https://go.dev/cl/207299 only marked it as supported on `arm64`)

**Comment by cherrymui** [comment]
Yeah, currently we only support internal linking on Android/Arm64.

We do support internal linking PIE on Linux/AMD64. I guess it would be too hard to make android/amd64 work as well.

We don't currently support internal linking PIE on 386 or 32-bit ARM. That would need more work. Are 32-bit Android platforms still important to support?

**Comment by ajsinclair** [comment]
> Are 32-bit Android platforms still important to support?

I would find it useful. We have Go binaries running on 32 bit ARM and x86 Emulators.
